### PR TITLE
Nstda thapanat

### PR DESCRIPTION
**Pull request by thapanat**
[EHN] Demo data for PABI2's Org Structure
[ADD] PABI2 Workflow configuration, User's profile, Project Memeber

### CODE REVIEW

File: pabi_project_member/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
```
Comment (pabi2) [inline pabi_project_member/__init__.py]: ตาม Coding Standard, เรามีแค่

# -*- coding: utf-8 -*-

2 บรรทัดเรื่องด้านล่าง (ในทุก file) ไม่ต้องมี

# © <YEAR(S)> <AUTHOR(S)>
# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).

File: pabi_workflow/models/wkf_config_purchase.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+from openerp import fields, models
+
+
+class WkfConfigPurchasingUnit(models.Model):
+    _name = 'wkf.config.pi.unit'
```
Comment (pabi2) [inline pabi_workflow/models/wkf_config_purchase.py]: I think "pi" do not give clear meaning. Should change to `wkf.config.purchase.unit`

File: pabi_workflow/models/wkf_config_purchase.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        'res.section',
+        'purchasinguint_section_rel',
+        'purchasing_unit_id',
+        'section_id',
+        string='Section',
+    )
+
+
+class WkfConfigPurchasingUnitResponsible(models.Model):
+    _name = 'wkf.config.pi.unit.responsible'
```
Comment (pabi2) [inline pabi_workflow/models/wkf_config_purchase.py]: I think "pi" do not give clear meaning. Should change to `wkf.config.purchase.unit.responsible`

Note: Python is dynamic language, all variable should have clear enough meaning.

File: pabi_project_member/models/res_project_member.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    _name = 'res.project.member'
+    _description = 'Project Member'
+
+    project_id = fields.Many2one(
+        'res.project',
+        string='Project',
+        required=True,
+        ondelete='cascade',
+    )
+    emp_id = fields.Many2one(
```
Comment (pabi2) [inline pabi_project_member/models/res_project_member.py]: For clear meaning, should be `employee_id`

File: pabi_project_member/models/res_project_member.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+
+class ResProject(models.Model):
+    _inherit = 'res.project'
+
+    member_ids = fields.One2many(
+        'res.project.member',
+        'project_id',
+        string='Project Member',
+        copy=False
```
Comment (pabi2) [inline pabi_project_member/models/res_project_member.py]: should be `copy=False,`

File: pabi_project_member/models/res_project_member.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        required=True,
+        ondelete='cascade',
+    )
+    emp_id = fields.Many2one(
+        'hr.employee',
+        string='Employee',
+        required=True,
+    )
+    project_position = fields.Selection([
+        ('PM', "Project Manager"),
```
Comment (pabi2) [inline pabi_project_member/models/res_project_member.py]: Selection should be,

```
[('manager', "Project Manager"),
 ('member', "Member"), ],
```

In Odoo, all selection is lower case, and give clear meaning.

File: pabi_workflow/models/wkf_config_base.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    _description = 'Work Flow Document Type Configuration'
+
+    name = fields.Char(
+        string='Name',
+        required=True,
+    )
+    description = fields.Text(
+        string='Description',
+    )
+    module = fields.Selection([
```
Comment (pabi2) [inline pabi_workflow/models/wkf_config_base.py]: - Procurement normally stand for "Purchase + Inventory", I think we can use it.
- In Odoo, "account" covers both Accounting and Finance

I think selection should be,

```
[('procure', "Purchase & Inventory"),
('account', "Accounting & Finance"), ],
```

File: pabi_workflow/models/wkf_config_base.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        'wkf.config.doctype',
+        string='Document Type',
+        required=True,
+    )
+    level = fields.Many2one(
+        'wkf.cmd.level',
+        string='Level',
+        required=True,
+    )
+    minimum = fields.Float(
```
Comment (pabi2) [inline pabi_workflow/models/wkf_config_base.py]: I think should be,

`amount_min / amount_max`

Idea is the name should gives clear meaning
(however, naming is also very subjective, please help decide)

File: pabi_workflow/models/wkf_config_boss.py [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+        string='Import Date',
+        readonly=True,
+    )
+
+
+class WkfCmdBossSpecailLevel(models.Model):
+    _name = 'wkf.cmd.boss.special.level'
+    _description = 'Boss Special Level Approval'
+
+    hr_employee_id = fields.Many2one(
```
Comment (pabi2) [inline pabi_workflow/models/wkf_config_boss.py]: Should be `employee_id` ?

File: pabi_workflow/models/wkf_config_project.py [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+class WkfCmdSpecialAmountProjectApproval(models.Model):
+    _name = 'wkf.cmd.special.amount.project.approval'
+    _description = 'Special Amount Project Approval'
+
+    doctype_id = fields.Many2one(
+        'wkf.config.doctype',
+        string='Document Type',
+        required=True,
+    )
+    emp_id = fields.Many2one(
```
Comment (pabi2) [inline pabi_workflow/models/wkf_config_project.py]: should be `employee_id` ?

File: pabi_workflow/models/wkf_config_project.py [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+        'hr.employee',
+        string='Employee',
+        required=True,
+    )
+    minimum = fields.Float(
```
Comment (pabi2) [inline pabi_workflow/models/wkf_config_project.py]: should be `amount_min` or `amount_minimum`?

File: pabi_workflow/models/wkf_config_project.py [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+        string="Minimum"
+    )
+    maximum = fields.Float(
```
Comment (pabi2) [inline pabi_workflow/models/wkf_config_project.py]: should be `amount_max` or `amount_maximum`?

File: pabi_workflow/models/wkf_config_purchase.py [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+        string='Org',
+        required=True,
+    )
+    doctype_id = fields.Many2one(
+        'wkf.config.doctype',
+        domain="[('module', '=', 'PI')]",
+        string='Document Type',
+        required=True,
+    )
+    responsible = fields.One2many(
```
Comment (pabi2) [inline pabi_workflow/models/wkf_config_purchase.py]: Coding standard:
should be, `responsible_ids`

File: pabi_workflow/models/wkf_config_purchase.py [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+        'wkf.config.pi.unit.responsible',
+        'purchasing_unit_id',
+        string='Responsible',
+        copy=True
+    )
+    section_id = fields.Many2many(
```
Comment (pabi2) [inline pabi_workflow/models/wkf_config_purchase.py]: should be `section_ids`

File: pabi_workflow/models/wkf_config_purchase.py [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+        'wkf.config.doctype',
+        domain="[('module', '=', 'PI')]",
+        string='Document Type',
+        required=True,
+    )
+    responsible = fields.One2many(
+        'wkf.config.pi.unit.responsible',
+        'purchasing_unit_id',
+        string='Responsible',
+        copy=True
```
Comment (pabi2) [inline pabi_workflow/models/wkf_config_purchase.py]: copy=True,

File: pabi_workflow/models/wkf_config_purchase.py [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+    _name = 'wkf.config.pi.unit.responsible'
+    _description = 'Purchasing Unit Responsible'
+
+    purchasing_unit_id = fields.Many2one(
+        'wkf.config.pi.unit',
+        string='Purchasing Unit',
+        required=True,
+        ondelete='cascade',
+    )
+    hr_employee_id = fields.Many2one(
```
Comment (pabi2) [inline pabi_workflow/models/wkf_config_purchase.py]: should be `employee_id`

File: pabi_workflow/models/wkf_config_base.py [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+        string="Minimum"
+    )
+    maximum = fields.Float(
+        string="Maximum"
```
Comment (pabi2) [inline pabi_workflow/models/wkf_config_base.py]: string="Maximum",

File: pabi_workflow/models/wkf_config_base.py [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+        string='Document Type',
+        required=True,
+    )
+    level = fields.Many2one(
+        'wkf.cmd.level',
+        string='Level',
+        required=True,
+    )
+    minimum = fields.Float(
+        string="Minimum"
```
Comment (pabi2) [inline pabi_workflow/models/wkf_config_base.py]: string="Minimum",

File: pabi_workflow/models/wkf_config_boss.py [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+        required=True,
+    )
+    org_id = fields.Many2one(
+        'res.org',
+        related='section_id.org_id',
+        string='Org',
+        readonly=True,
+        store=True,
+    )
+    hr_employee_id = fields.Many2one(
```
Comment (pabi2) [inline pabi_workflow/models/wkf_config_boss.py]: should be `employee_id`

File: pabi_workflow/models/wkf_config_purchase.py [hunk 20]
```diff
@@ -0,0 +1,62 @@
...
+        string='Description',
+    )
+    org_id = fields.Many2one(
+        'res.org',
+        string='Org',
+        required=True,
+    )
+    doctype_id = fields.Many2one(
+        'wkf.config.doctype',
+        domain="[('module', '=', 'PI')]",
```
Comment (pabi2) [inline pabi_workflow/models/wkf_config_purchase.py]: "PI" is changed?

File: pabi_project_member/__init__.py (hunk 1 above)
Comment (thapanat) [inline pabi_project_member/__init__.py]: I copied from your code. 
please, remove too.

File: pabi_workflow/models/wkf_config_purchase.py (hunk 2 above)
Comment (thapanat) [inline pabi_workflow/models/wkf_config_purchase.py]: I solved it.

File: pabi_workflow/models/wkf_config_purchase.py (hunk 3 above)
Comment (thapanat) [inline pabi_workflow/models/wkf_config_purchase.py]: I solved it.

File: pabi_project_member/models/res_project_member.py (hunk 4 above)
Comment (thapanat) [inline pabi_project_member/models/res_project_member.py]: I solved it.

File: pabi_project_member/models/res_project_member.py (hunk 5 above)
Comment (thapanat) [inline pabi_project_member/models/res_project_member.py]: I solved it.

File: pabi_project_member/models/res_project_member.py (hunk 6 above)
Comment (thapanat) [inline pabi_project_member/models/res_project_member.py]: I solved it.

File: pabi_workflow/models/wkf_config_base.py (hunk 7 above)
Comment (thapanat) [inline pabi_workflow/models/wkf_config_base.py]: I solved it.
I try to define short name for each module.
but I think same your idea. it easy to understand for next developer.

File: pabi_workflow/models/wkf_config_base.py (hunk 8 above)
Comment (thapanat) [inline pabi_workflow/models/wkf_config_base.py]: I solved it.

File: pabi_workflow/models/wkf_config_project.py (hunk 12 above)
Comment (thapanat) [inline pabi_workflow/models/wkf_config_project.py]: I solved it.

File: pabi_workflow/models/wkf_config_purchase.py (hunk 13 above)
Comment (thapanat) [inline pabi_workflow/models/wkf_config_purchase.py]: I solved it.

File: pabi_workflow/models/wkf_config_purchase.py (hunk 14 above)
Comment (thapanat) [inline pabi_workflow/models/wkf_config_purchase.py]: I solved it.

File: pabi_workflow/models/wkf_config_purchase.py (hunk 15 above)
Comment (thapanat) [inline pabi_workflow/models/wkf_config_purchase.py]: I solved it.

File: pabi_workflow/models/wkf_config_purchase.py (hunk 16 above)
Comment (thapanat) [inline pabi_workflow/models/wkf_config_purchase.py]: I solved it.

File: pabi_workflow/models/wkf_config_base.py (hunk 17 above)
Comment (thapanat) [inline pabi_workflow/models/wkf_config_base.py]: I solved it.

File: pabi_workflow/models/wkf_config_base.py (hunk 18 above)
Comment (thapanat) [inline pabi_workflow/models/wkf_config_base.py]: I solved it.

File: pabi_workflow/models/wkf_config_boss.py (hunk 19 above)
Comment (thapanat) [inline pabi_workflow/models/wkf_config_boss.py]: I solved it.

File: pabi_workflow/models/wkf_config_purchase.py (hunk 20 above)
Comment (thapanat) [inline pabi_workflow/models/wkf_config_purchase.py]: I solved, it change to "procure".

File: pabi_workflow/models/wkf_config_project.py (hunk 11 above)
Comment (thapanat) [inline pabi_workflow/models/wkf_config_project.py]: I solved it.

File: pabi_workflow/models/wkf_config_project.py (hunk 10 above)
Comment (thapanat) [inline pabi_workflow/models/wkf_config_project.py]: I solved it.

File: pabi_workflow/models/wkf_config_boss.py (hunk 9 above)
Comment (thapanat) [inline pabi_workflow/models/wkf_config_boss.py]: ใน Module : HR employee ผมเพิ่ม Filed employee_id สำหรับเก็บรหัสพนักงาน (Employee Id.) เมื่อเรียกใช้จาก module อื่นกลัวจะไปซ้ำกัน ก็เลยตั้งชื่อ field : hr_employee_id  

ผมแก้ไขโดย ที่  Module : HR employee field employee_id  แก้เป็น employee_code ดีกว่าครับ

**Pull request merged**